### ISSUE TITLE
Maintainance of pic8259_simple crate

### ISSUE THREAD

**Issue by phil-opp** [emk/toyos-rs#5]
Hi @emk! I plan to finally write a post about hardware interrupts for https://os.phil-opp.com/. Your `pic8259_simple` crate would be very useful for this. Thanks for writing it!

One concern I have is that this might prompt people to file issues or submit pull requests, so I wanted to ask whether you're still interested in maintaining this crate. If yes, awesome! If not, we could maintain it under the [rust-osdev](https://github.com/rust-osdev/) organization if you like (only the pic crate, not toyos). Or we could fork it and publish it under a new name, if you prefer.

Thanks again!

**Comment by emk** [comment]
I loved your OS series and it was a huge inspiration to me!

I'm definitely happy to accept PRs, though I'm in the middle of moving to a new house for the next few weeks, so turnaround might be a bit spotty until late August.

I'd also be willing to share maintenance of this crate under the the rust-osdev organization, provided I still keep commit access. :-) That way you could make new releases whenever you wanted.

Note that pic8259_simple also relies on [cpuio](https://github.com/emk/toyos-rs/tree/master/crates/cpuio), a very simple and (I hope) clean interface to `outb` and friends. This can also move to rust-osdev if that makes sense for you.

Basically, I'm happy to do anything I can do to make these crates more usable for your readers! If you have API feedback, etc., we can definitely change things.

**Comment by phil-opp** [comment]
> I loved your OS series and it was a huge inspiration to me!

Thank you so much, really great to hear that!

> I'm definitely happy to accept PRs, though I'm in the middle of moving to a new house for the next few weeks, so turnaround might be a bit spotty until late August.

Ok cool! Then I would say let's keep things the way they are for now. I'm quite happy with the API and documentation, so I think I will just use the current crates.io version.

I sent you an invitation to rust-osdev, so you can always share this crate if you become overwhelmed with maintainance. You would of course keep admin access.

> Note that pic8259_simple also relies on cpuio, a very simple and (I hope) clean interface to outb and friends. This can also move to rust-osdev if that makes sense for you.

We also have a simple [port interface](https://docs.rs/x86_64/0.2.11/x86_64/instructions/port/index.html) as part of the `x86_64` crate. Seems like `cpuio` is a bit more featureful (`Port` vs `UnsafePort`), but I only took a quick look. I'll definitely keep it in mind, maybe it makes sense to use it in `x86_64`.

> Basically, I'm happy to do anything I can do to make these crates more usable for your readers! If you have API feedback, etc., we can definitely change things.

Thanks so much for your help! I'm quite happy with the current API and I think your crate will be very useful for my readers. Thanks again for creating it!

**Comment by phil-opp** [comment]
Hi @emk, are you still willing to maintain this crate? I saw that you didn't reply in https://github.com/emk/toyos-rs/pull/7 for the last months, so I thought I'd ask again. I'd be happy to move this crate to the `rust-osdev` organization if you like some help in maintainence. If you prefer, we could also create a fork of it if you prefer.

**Comment by phil-opp** [comment]
As noted in https://github.com/emk/toyos-rs/pull/7#issuecomment-842215920, I created a fork of the `pic8259` crate under the `rust-osdev` organization: https://github.com/rust-osdev/pic8259. The forked crate is avaliable on crates.io as `pic8259` v0.10.1.
